File: Soultion/Count-Good-Numbers.java
class Solution {
     long mod=1000000007;
    public int countGoodNumbers(long n) {
         
         return (int)((binaryExp(5,(n+1)/2))*binaryExp(4,(n/2))%mod);

    }
    public long binaryExp(int x, long y){
        long mul=x;
        long result=1;
        while(y>0){
            if((y&1)==1){
                result=(result*mul)%mod;
            }
            mul=(mul*mul)%mod;
            y/=2;
        }
        return result;
    }
}

File: Soultion/Count-Good-Numbers.py
class Solution:
    def countGoodNumbers(self, n: int) -> int:
        mod = 10**9 + 7

        def binaryExp(x, y):
            res = 1
            mul = x % mod

            while y > 0:

                if y % 2 == 1:
                    res = (res * mul) % mod

                mul = (mul * mul) % mod
                y //= 2
            return res

        even_pos = (n + 1) // 2
        odd_pos = n // 2

        return (binaryExp(5, even_pos) * binaryExp(4, odd_pos)) % mod
